src/multiplexed-session.ts

```diff
@@ -0,0 +1,320 @@
+        ) {
+          return;
+        }
+        this.emit('error', err);
+      });
+  }
+
+  /**
+   * Creates a new multiplexed session.
+   *
+   * This method sends a request to the database to create a new session with multiplexing enabled.
+   * The response from the database would be an array, the first value of the array will be containing the multiplexed session.
+   *
+   * @returns {Promise<void>} A Promise that resolves when the session has been successfully created and assigned, an event
+   * `mux-session-available` will be emitted to signal that the session is ready.
+   *
+   * @private
+   */
+  async _createSession(): Promise<void> {
+    try {
+      const [createSessionResponse] = await this.database.createSession({
+        multiplexed: true,
+      });
+      this._muxSession.multiplexedSession = createSessionResponse;
+      this.emit('mux-session-available');
+    } catch (e) {
+      this.emit('error', e);
+    }
+  }
+
+  /**
+   * Maintains the multiplexed session by periodically refreshing it.
+   *
+   * This method sets up a periodic refresh interval for maintaining the session. The interval duration
+   * is determined by the @param refreshRate option, which is provided in minutes.
+   * The default value is 7 days.
+   *
+   * @returns {void} This method does not return any value.
+   *
+   * @throws {Error} If the `refreshRate` option is not defined, this method could throw an error.
+   */
+  _maintain(): void {
+    const refreshRate = this.options.refreshRate! * 24 * 60 * 60000;
+    this._refreshHandle = setInterval(() => this._refresh(), refreshRate);
+    this._refreshHandle.unref();
+  }
+
+  /**
+   * Creates a transaction for a session.
+   *
+   * @private
+   *
+   * @param {Session} session The session object.
+   * @param {object} options The transaction options.
+   */
+  _prepareTransaction(session: Session | null): void {
+    const transaction = session!.transaction(
+      (session!.parent as Database).queryOptions_
+    );
+    session!.txn = transaction;
+  }
+
+  /**
+   * Refreshes the session by checking its expiration time and recreating the session if expired.
+   * Every 7th day a new Multiplexed Session will get created. However, a Multiplexed Session will
+   * be alive for 30 days in the backend.
+   *
+   * @returns {Promise<void>} A Promise that resolves once the refresh process is completed.
+   *                          If the session is expired, a new session will be created.
+   *
+   * @throws {Error} If there is an issue with retrieving the session metadata or calculating the expiration time.
+   */
+  async _refresh(): Promise<void> {
+    const metadata = await this._muxSession.multiplexedSession?.getMetadata();
+    const createTime =
+      parseInt(metadata![0].createTime.seconds) * 1000 +
+      metadata![0].createTime.nanos / 1000000;
+
+    // Calculate expiration time (7 days after session creation)
+    const expireTime = createTime + 7 * 24 * 60 * 60 * 1000;
```

Comment:
Instead of 7, we should be using `refreshRate` ?

Reply:
Updated, thanks!